src/divideIntoSections.js

```diff
@@ -0,0 +1,22 @@
+// split by double-newline, but preserve HTML blocks as single sections:
+module.exports = function divideIntoSections(content) {
+  var sections = [];
+  content = content.replace(/[\n]{2,}/g, "\n\n"); // manage double newlines (not sure exactly?)
+
+  // Split by <el>...</el> blocks, then parse the markdown text between it.
+  // If we don't do this, it'll fail if there are double newlines inside HTML blocks, esp. <script> blocks.
+
+  // test string: s = "<div>lala</div>\n\nhey<table class='hey'><p><table></table></p></table>\n\n## Markdown\n\n<p>Hi there</p>\n\n* One\n* Two"
+
+  var jqueryCollection = $(content);
+
+  jqueryCollection.toArray().forEach(function(chunk) {
+    if (chunk.nodeName === "#text") {
+      sections.concat(chunk.split("\n\n")); // split by double newline and add
+    } else {
+      sections.push(chunk.toString()); // it's an HTML chunk
+    }
+  });
+
+  return sections;
```

Comment:
We also need to strip out empty list items from sections here!

Reply:
At L28 I suppose we can,
```js
sections = sections.filter(function (x) {
  return (x != null || x!="");
});
```

Reply:
This didn't seem to work, it's still looking for 13... maybe they're not totally empty? Shall we check if they're whitespace? ` /\s/` maybe?

Reply:
I tried `x.match(/\S/) !== nil` as one more in case there is whitespace.